test/integrations/destinations/algolia/processor/data.ts

```diff
@@ -1417,4 +1417,214 @@ export const data = [
+            statTags: {
+              destType: 'ALGOLIA',
+              errorCategory: 'dataValidation',
+              errorType: 'instrumentation',
+              feature: 'processor',
+              implementation: 'cdkV2',
+              module: 'destination',
+              destinationId: 'destId',
+              workspaceId: 'wspId',
+            },
+            statusCode: 400,
+            metadata: {
+              destinationId: 'destId',
+              workspaceId: 'wspId',
+            },
+          },
+        ],
+      },
+    },
+  },
+  {
+    name: 'algolia',
+    description: 'filters or objectIds must be non empty',
+    feature: 'processor',
+    module: 'destination',
+    version: 'v0',
+    input: {
+      request: {
+        body: [
+          {
+            message: {
+              type: 'track',
+              event: 'product clicked',
+              sentAt: '2024-02-25T17:55:36.882Z',
+              userId: '12345',
+              channel: 'web',
+              properties: {
+                index: 'products',
+                list_id: 'search_results_page',
+                queryId: '8e737',
+                products: [],
+                eventName: 'productListView',
+                list_name: 'Search Results Page',
+                objectIds: [],
+                positions: [],
+                userToken: 'e494',
+                additional_attributes: {},
+              },
+              receivedAt: '2024-02-25T17:55:38.089Z',
+              request_ip: '107.130.37.100',
+              anonymousId: '68e9f4b8-fd4d-4c56-8ca4-858de2fd1df8',
+              integrations: {
+                All: true,
+              },
+              originalTimestamp: '2024-02-25T17:55:36.880Z',
+            },
+            destination: {
+              DestinationDefinition: {
+                Config: {
+                  cdkV2Enabled: true,
+                  excludeKeys: [],
+                  includeKeys: [],
+                },
+              },
+              Config: {
+                apiKey: 'dummyApiKey',
+                applicationId: 'O2YARRI15I',
+                eventTypeSettings: [
+                  {
+                    from: 'product clicked',
+                    to: 'cLick ',
+                  },
+                ],
+              },
+            },
+            metadata: {
+              destinationId: 'destId',
+              workspaceId: 'wspId',
+            },
+          },
+        ],
+      },
+    },
+    output: {
+      response: {
+        status: 200,
+        body: [
+          {
+            error:
+              'Either filters or objectIds is required and must be non empty.: Workflow: procWorkflow, Step: validateDestPayload, ChildStep: undefined, OriginalError: Either filters or objectIds is required and must be non empty.',
+            statTags: {
```

Comment:
can we reuse the statTags

Reply:
doing them separate as part of component test rerstructure